lib/floki/finder.ex

```diff
@@ -95,7 +95,12 @@ defmodule Floki.Finder do
           nil -> [n|res_acc]
           _ ->
             {_, _, children_nodes} = n
-            traverse(children_nodes, sibling_nodes, selector.combinator.selector, res_acc)
+
+            if selector.combinator.match_type == :child do
+              traverse_child(children_nodes, sibling_nodes, selector.combinator.selector, res_acc)
+            else
+              traverse(nodes, sibling_nodes, selector, res_acc)
```

Comment:
@gmile I'm still thinking about this. Not sure what is going wrong. One suggestion is to isolate that `case` statement from the line 70, and try to reuse that here.

I suspect that something is wrong with the way `sibling_nodes` are traversed (this explain why the `div > p > span > img + img` is not working).

I'm gonna check this tomorrow night again. Thanks!